src/jit/lowerxarch.cpp

```diff
@@ -2015,16 +2064,21 @@ Lowering::TreeNodeInfoInitModDiv(GenTree* tree)
         {   
             // No implicit conversions at this stage as the expectation is that
             // everything is made explicit by adding casts.
-            assert(tree->gtOp.gtOp1->TypeGet() == tree->gtOp.gtOp2->TypeGet());
+            assert(op1->TypeGet() == op2->TypeGet());
 
-            info->srcCount = 2;
-            info->dstCount = 1;
-
-            GenTree* op2 = tree->gtOp.gtOp2;
             if (op2->isMemoryOp() || op2->IsCnsNonZeroFltOrDbl())
             {
                 MakeSrcContained(tree, op2);
             }
+            else
+            {
+                // If there are no containable memory operands, mark which of the
+                // operands can be treated as reg optional. If no register gets
+                // allocated to such an operand, it will be treated as a contained
+                // memory operand during codegen.
+                SetRegOptionalForBinOp(tree);
```

Comment:
I don't think it's necessary to have such a verbose comment repeated everywhere. Perhaps "If there are no containable operands, we can make an operand reg-optional." I think it should be clear enough that it would be treated as contained if it didn't get a register. Then, you can leave the more verbose explanation for the `SetRegOptionalForBinOp` method.